src/features/change-bounds/movement-restrictor.ts

```diff
@@ -14,276 +14,76 @@
-        const targetLeftLine = new PointToPointLine(targetTopLeft, targetBottomLeft);
-        const targetRightLine = new PointToPointLine(targetTopRight, targetBottomRight);
-
-        // Compute distances
-        const distanceTop = this.getDistanceBetweenParallelLines(elementBottomLeft, elementBottomRight, targetTopLine);
-        const distanceBottom = this.getDistanceBetweenParallelLines(elementTopLeft, elementTopRight, targetBottomLine);
-        const distanceLeft = this.getDistanceBetweenParallelLines(elementTopLeft, elementBottomLeft, targetRightLine);
-        const distanceRight = this.getDistanceBetweenParallelLines(elementTopRight, elementBottomRight, targetLeftLine);
-
-        const minimumCandidates: number[] = [];
-
-        // Overlap on the horizontal lines
-        if (isOverlapping1Dimension(element.bounds.x, element.bounds.width, target.bounds.x, target.bounds.width)) {
-            minimumCandidates.push(distanceTop.valueOf());
-            minimumCandidates.push(distanceBottom.valueOf());
-        }
-        // Overlap on the horizontal lines
-        if (isOverlapping1Dimension(element.bounds.y, element.bounds.height, target.bounds.y, target.bounds.height)) {
-            minimumCandidates.push(distanceLeft.valueOf());
-            minimumCandidates.push(distanceRight.valueOf());
-        }
-
-        // Get minimum distance and then snap accordingly
-        minimumCandidates.sort((a, b) => a - b);
-        const minimumDistance = minimumCandidates[0];
+    cssClasses = ["movement-not-allowed"];
+}
 
-        if (minimumDistance === distanceTop) {
-            snappedBounds = {
-                x: element.bounds.x,
-                y: target.bounds.y - 1 - element.bounds.height,
-                width: element.bounds.width,
-                height: element.bounds.height
-            };
-        }
-        if (minimumDistance === distanceBottom) {
-            snappedBounds = {
-                x: element.bounds.x,
-                y: target.bounds.y + target.bounds.height + 1,
-                width: element.bounds.width,
-                height: element.bounds.height
-            };
-        }
-        if (minimumDistance === distanceLeft) {
-            snappedBounds = {
-                x: target.bounds.x + target.bounds.width + 1,
-                y: element.bounds.y,
-                width: element.bounds.width,
-                height: element.bounds.height
-            };
-        }
-        if (minimumDistance === distanceRight) {
-            snappedBounds = {
-                x: target.bounds.x - 1 - element.bounds.width,
-                y: element.bounds.y,
-                width: element.bounds.width,
-                height: element.bounds.height
-            };
-        }
-        return snappedBounds;
-    }
+export function doOverlap(element1: SModelElement & BoundsAware, element2: SModelElement & BoundsAware) {
```

Comment:
Probably just a matter of taste, but I'd prefer `areOverlapping`, because `doOverlap` doesn't sound to me like a boolean-query ("is", "does", etc.) but rather like an operation with side-effect.